### DISCUSSION TITLE
Same individual growth rates

### DISCUSSION THREAD

**Discussion by lhtxa** [micom-dev/micom#132] [Q&A] [answered]
Hi! Is there a way to find a solution where some, or all, individual growth rates have the same value?

**Comment by cdiener** [comment, marked as answer]
Definitely possible but currently a bit clunky. Cooperative Tradeoff will push the solution towards that by default, especially if you choose a low tradeoff. But it is also possible to enforce the same growth rates directly by adding ratio constraints. Here is an example:

```python3
import micom as mm

com = mm.Community(mm.data.test_taxonomy())
constraints = []
ref_taxon_objective = com.constraints["objective_" + com.taxa[0]].expression
for tax in com.taxa[1:]:
    c = com.problem.Constraint(
        com.constraints["objective_" + tax].expression - ref_taxon_objective,
        name="ratio_" + tax,
        lb=0, ub=0
    )
    constraints.append(c)
with com:
    com.add_cons_vars(constraints)
    sol = com.optimize()   # maximizes community growth with the constraints applied

print(sol.members)
```

Which will give:

```
                    abundance  growth_rate  reactions  metabolites
compartments                                                      
Escherichia_coli_1       0.25     0.873922         95           72
Escherichia_coli_2       0.25     0.873922         95           72
Escherichia_coli_3       0.25     0.873922         95           72
Escherichia_coli_4       0.25     0.873922         95           72
medium                    NaN          NaN         20           20
```

So this is [planned to be implemented](https://github.com/orgs/micom-dev/projects/3/views/1) as a helper function already.

**Reply by lhtxa** [reply]
Thanks for the reply with the code! 
Are the results expected to be the same whatever is the member defined as ref_taxon? I varied it and did get the same results.

**Reply by cdiener** [reply]
Yes, reference taxon does not matter.
